components/Chart.js

```diff
@@ -2,33 +2,39 @@ import React, { useState } from "react";
 import styled from "styled-components";
 import { BarChart, YAxis, XAxis, Bar } from "recharts";
 import useSWR from "swr";
+import { capitalizeFirstLetter } from "@/lib/utils";
 
 export default function BarChartPage() {
   const { data: transactionsList = [] } = useSWR("/api/transactions");
   const [selectedCategory, setSelectedCategory] = useState(null);
+  const [transactionType, setTransactionType] = useState("income");
+
+  const totalByCategory = (type) =>
+    transactionsList
+      .filter((transaction) => transaction.type === type)
+      .reduce((accumulator, transaction) => {
+        const categoryName = transaction.category?.name || "Uncategorized";
+        accumulator[categoryName] =
+          (accumulator[categoryName] || 0) + transaction.amount;
+        return accumulator;
+      }, {});
+
+  const getChartData = () => {
+    const data = totalByCategory(transactionType);
+    return Object.entries(data)
+      .map(([name, value]) => ({
+        name,
+        value,
+        fill: "var(--text-color-dark)",
+      }))
+      .sort((a, b) => b.value - a.value);
+  };
 
-  const totalExpensesByCategory = transactionsList
-    .filter((transaction) => transaction.type === "expense")
-    .reduce((accumulator, transaction) => {
-      const categoryName = transaction.category?.name || "Uncategorized";
-
-      accumulator[categoryName] =
-        (accumulator[categoryName] || 0) + transaction.amount;
-
-      return accumulator;
-    }, {});
-
-  const chartData = Object.entries(totalExpensesByCategory)
-    .map(([name, value]) => ({
-      name,
-      value,
-      fill: "var(--text-color-dark)",
-    }))
-    .sort((a, b) => b.value - a.value);
+  const chartData = getChartData();
 
-  const renderCustomLabel = ({ x, y, width, value }) => {
+  const renderCustomLabel = ({ x, y, width, height, value }) => {
     return (
-      <StyledLabelText x={x + width + 10} y={y + 22}>
+      <StyledLabelText x={x + width + 10} y={y + height / 2 + 5}>
         {value} €
       </StyledLabelText>
```

Comment:
This is a component. You should not define components inside other components unless its absolutely necessary. Please move this below the end of the function "BarChartPage" next to the styled components.